src/hooks/useQuiz.ts

```diff
@@ -22,11 +23,17 @@ const useQuiz: UseQuiz = (quizOptions) => {
   const quizEvents = useQuizEvents(quizOptions, cioClient, quizState);
 
   // Props getters
-  const { quizApiState, quizLocalState, skipToResults } = quizState;
+  const { quizApiState, quizLocalState, quizSessionStorageState } = quizState;
+  const { skipToResults } = quizSessionStorageState;
   const propGetters = usePropsGetters(quizEvents, quizApiState, quizLocalState);
 
   const primaryColorStyles = usePrimaryColorStyles(primaryColor);
 
+  useEffect(() => {
+    if (skipToResults) quizEvents.hydrateQuiz();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
```

Comment:
Always hydrate results if skipToResults is true

Reply:
skipToResults is true when

- Hydration is enabled in the config object
- The quiz is completed
- And showSessionModalOnResults is false